src/Query/Grammars/CypherGrammar.php

```diff
@@ -871,8 +871,9 @@ public function compileMatchRelationship(Builder $query, $attributes)
             $endKey = $attributes['end']['id']['key'];
             $endNode = 'rel_'.$this->modelAsNode($attributes['label']);
             $endLabel = $this->prepareLabels($attributes['end']['label']);
+            $endId = null;
 
-            if ($attributes['end']['id']['value']) {
+            if ($attributes['end']['id']['value'] || $attributes['end']['id']['value'] === 0) {
```

Comment:
thanks for this fix, it's a gotcha! I don't know how we've gone so long without facing such issue 😅

Do you think we could do `if (isset($attributes['end']['id']['value'])` instead of the two checks? Haven't had the chance to try it yet sorry.

Reply:
Well, if you have a running database it's unlikely to delete the very first node after a while. :)

I couldn't find the usecase when this is called without a value attribute, but I didn't want to break it anyhow. It's not absolutely clear how it is used, because this is already in an `if (isset($attributes['end']))` check (line 870).

Also the current released check is also not completely notice free, because a `$attribute['end']['id']` without a `value` key would throw an `undefined index notice` on the check.

So my best bet is, that it is unlikely that this method gets called in the way it is checked.